java/vector/src/main/java/org/apache/arrow/vector/IntVector.java

```diff
@@ -344,8 +347,11 @@ public TransferPair makeTransferPair(ValueVector to) {
   }
 
   @Override
-  public void setEncodedValue(int index, int value) {
-    this.setSafe(index, value);
+  public void setWithPossibleTruncate(int index, long value) {
+    if (value > Integer.MAX_VALUE) {
```

Comment:
sorry for the misunderstanding lets just set the value directly no warning.

Reply:
ah, fixed.